marlowe-playground-client/src/HaskellEditor/View.purs

```diff
@@ -0,0 +1,162 @@
+        [ HTML.id_ "editor-options"
+        , class_ (ClassName "dropdown-header")
+        , onSelectedIndexChange (\idx -> ChangeKeyBindings <$> toEnum idx)
+        ]
+        (map keybindingItem (upFromIncluding bottom))
+    ]
+  where
+  keybindingItem item =
+    if state ^. _haskellEditorKeybindings == item then
+      option [ class_ (ClassName "selected-item"), HTML.value (show item) ] [ text $ show item ]
+    else
+      option [ HTML.value (show item) ] [ text $ show item ]
+
+haskellEditor ::
+  forall m.
+  MonadAff m =>
+  State ->
+  ComponentHTML Action ChildSlots m
+haskellEditor state = slot _haskellEditorSlot unit component unit (Just <<< HandleEditorMessage)
+  where
+  setup editor =
+    liftEffect do
+      mContents <- LocalStorage.getItem StaticData.bufferLocalStorageKey
+      let
+        contents = fromMaybe HE.escrow mContents
+      model <- Monaco.getModel editor
+      Monaco.setValue model contents
+
+  component = monacoComponent $ HM.settings setup
+
+bottomPanel :: forall p. State -> HTML p Action
+bottomPanel state =
+  div
+    ( [ classes
+          ( if showingBottomPanel then
+              [ analysisPanel ]
+            else
+              [ analysisPanel, collapsed ]
+          )
+      , bottomPanelHeight showingBottomPanel
+      ]
+    )
+    [ div
+        [ classes [ footerPanelBg, ClassName "flip-x" ] ]
+        [ section [ classes [ ClassName "panel-header", aHorizontal ] ]
+            [ div [ classes [ ClassName "panel-sub-header-main", aHorizontal ] ]
+                [ div [ class_ (ClassName "minimize-icon-container") ]
+                    [ a [ onClick $ const $ Just $ ShowBottomPanel (state ^. _showBottomPanel <<< to not) ]
+                        [ img [ classes (minimizeIcon $ state ^. _showBottomPanel), src closeDrawerArrowIcon, alt "close drawer icon" ] ]
+                    ]
+                ]
+            ]
+        , section
+            [ classes [ ClassName "panel-sub-header", aHorizontal, ClassName "panel-contents" ]
+            ]
+            (resultPane state)
+        ]
+    ]
+  where
+  showingBottomPanel = state ^. _showBottomPanel
+
+compileButton :: forall p. State -> HTML p Action
+compileButton state = button [ onClick $ const $ Just Compile ] [ text (if state ^. _compilationResult <<< to isLoading then "Compiling..." else "Compile") ]
+
+sendResultButton :: forall p. State -> String -> Action -> HTML p Action
+sendResultButton state msg action =
+  let
+    compilationResult = view _compilationResult state
+  in
+    case view _compilationResult state of
+      Success (Right (InterpreterResult result)) ->
+        button
+          [ onClick $ const $ Just action
+          , disabled (isLoading compilationResult || (not isSuccess) compilationResult)
+          ]
+          [ text msg ]
+      _ -> text ""
+
+resultPane :: forall p. State -> Array (HTML p Action)
+resultPane state =
+  if state ^. _showBottomPanel then case view _compilationResult state of
```

Comment:
I think `case a, b of` would be more appropriate here.

Reply:
oh yeah, I often forget about this one too, it's nice.